numcodecs/bitround.py

```diff
@@ -0,0 +1,31 @@
+import numpy as np
+
+
+from .abc import Codec
+from .compat import ensure_ndarray, ndarray_copy
+
+
+class BitRound(Codec):
+    codec_id = 'bitround'
+
+    def __init__(self, keepbits: int):
+        self.keepbits = keepbits
+
+    def encode(self, buf):
+        # TODO: figure out if we need to make a copy
+        # Currently this appears to be overwriting the input buffer
+        # Is that the right behavior?
```

Comment:
In BitInformation.jl the rounding is implemented as scalar version which does not overwrite the input (as float32 is immutable so a copy is created anyway), however, I define a rounding function for arrays, that can either act in-place (i.e. overwriting the bits in an existing array) or acts on a copy of the array, such that the input array in unchanged.

Reply:
No we shouldn't be overwriting the input buffer.

Reply:
How about changing
```
b = a.view(dtype=np.int32)
```
to
```
b = a.view(dtype=np.int32).copy()
```
to avoid the overwriting of the input buffer?

Reply:
`a.astype(np.int32, copy=True)` is more canonical.

Reply:
Continued in PR: https://github.com/zarr-developers/numcodecs/pull/608